monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -1,29 +1,58 @@
+import functools
 import logging
 
 import requests
 
-from common.common_consts.timeouts import MEDIUM_REQUEST_TIMEOUT
+from common.common_consts.timeouts import LONG_REQUEST_TIMEOUT, MEDIUM_REQUEST_TIMEOUT
 
 from . import IIslandAPIClient, IslandAPIConnectionError, IslandAPIError, IslandAPITimeoutError
 
 logger = logging.getLogger(__name__)
 
 
-class HTTPIslandAPIClient(IIslandAPIClient):
-    """
-    A client for the Island's HTTP API
-    """
-
-    def __init__(self, island_server: str):
+def handle_island_errors(fn):
+    @functools.wraps(fn)
+    def decorated(*args, **kwargs):
         try:
-            requests.get(  # noqa: DUO123
-                f"https://{island_server}/api?action=is-up",
-                verify=False,
-                timeout=MEDIUM_REQUEST_TIMEOUT,
-            )
+            return fn(*args, **kwargs)
         except requests.exceptions.ConnectionError as err:
             raise IslandAPIConnectionError(err)
         except TimeoutError as err:
             raise IslandAPITimeoutError(err)
         except Exception as err:
             raise IslandAPIError(err)
+
+    return decorated
+
+
+class HTTPIslandAPIClient(IIslandAPIClient):
+    """
+    A client for the Island's HTTP API
+    """
+
+    @handle_island_errors
+    def __init__(self, island_server: str):
+        requests.get(  # noqa: DUO123
+            f"https://{island_server}/api?action=is-up",
+            verify=False,
+            timeout=MEDIUM_REQUEST_TIMEOUT,
+        )
+
+    @staticmethod
+    @handle_island_errors
+    def send_log(server_address: str, data: str):
```

Comment:
This class inherits from `IIslandAPIClient`. `IIslandAPIClient` should contain the API for communicating with the Island. Therefore, we should have stubs for these methods in the `IIslandAPIClient` and document those methods.

I don't think these methods should be static, because we can't guarantee that other implementations of `IIslandAPIClient` will be able to implement those methods statically.